msg_sequence/test_pairing.py

```diff
@@ -2,100 +2,89 @@
 Test for 'Paired request missing, HTTP Response Splitting attack?' error
 """
 
-from __future__ import print_function
-
-from helpers import chains, deproxy, dmesg, tempesta, tf_cfg
-from testers import functional
-
 __author__ = "Tempesta Technologies, Inc."
 __copyright__ = "Copyright (C) 2018 Tempesta Technologies, Inc."
 __license__ = "GPL2"
 
-
-class PairingTest(functional.FunctionalTest):
-    chains_size = 2
-    send_to_close = chains_size / 2
-    defconfig = ""
-
-    def create_servers(self):
-        port = tempesta.upstream_port_start_from()
-        self.servers = [deproxy.Server(port=port, conns_n=1)]
-
-    def create_tester(self):
-        message_chains = [chains.proxy() for _ in range(self.chains_size)]
-        self.tester = PairingTester(self.client, self.servers, message_chains=message_chains)
-
-    def prepare(self):
-        self.tempesta.config.set_defconfig(self.defconfig)
-        self.configure_tempesta()
-
-        for server in self.servers:
-            server.start()
-
-        self.tempesta.start()
-        self.client.start()
-        self.tester.start()
-
+from framework import tester
+from framework.parameterize import parameterize_class
+from helpers import dmesg, util
+
+
+DEPROXY_CLIENT = {
+    "id": "deproxy",
+    "type": "deproxy",
+    "addr": "${tempesta_ip}",
+    "port": "80",
+}
+
+DEPROXY_CLIENT_H2 = {
+    "id": "deproxy",
+    "type": "deproxy_h2",
+    "addr": "${tempesta_ip}",
+    "port": "443",
+    "ssl": True,
+}
+
+
+@parameterize_class(
+    [
+        {"name": "Http", "clients": [DEPROXY_CLIENT]},
+        {"name": "H2", "clients": [DEPROXY_CLIENT_H2]},
+    ]
+)
+class PairingTest(tester.TempestaTest):
+    tempesta = {
+        "config": """
+listen 80;
+listen 443 proto=h2;
+
+server ${server_ip}:8000;
+
+vhost default {
+    proxy_pass default;
+    tls_certificate ${tempesta_workdir}/tempesta.crt;
+    tls_certificate_key ${tempesta_workdir}/tempesta.key;
+    tls_match_any_server_name;
+}
+""",
+    }
+
+    backends = [
+        {
+            "id": "deproxy",
+            "type": "deproxy",
+            "port": "8000",
+            "response": "static",
+            "response_content": "",
```

Comment:
```suggestion
            "response_content": "HTTP/1.1 200 OK\r\n",
```
I think you should use an incomplete response here, otherwise Tempesta will not receive data from the deproxy server. I was expected here - TCP PSH + ACK from deproxy server, but I received just TCP ACK.